### PR TITLE
Fix an error in Http Protocol when data content is null

### PR DESCRIPTION
**Pull request by ThomasF34**
## What does this PR do ?

When a controller action is returning a raw content but this content is `null`, an error is thrown since we are trying to access `content.type`. 

This PR transforms the verification `data.raw` into `data.raw && data.content`. Thus, if the content is null we will now check the type of it. (and directly go to the stringify step) 

### How should this be manually tested?

- Step 1 : in a controller action : 
```js 
async anAction(request){
  request.response.raw = true 
  return null
}
```
- Step 2 : when you're calling this action, the error is thrown

### CODE REVIEW

**Comment by scottinet** [comment]
Also, I disagree with how this bug is fixed: the new condition makes Kuzzle think that the data is NOT raw, and so it embeds the null result in a standard Kuzzle API response, which is not the intended behavior.

Instead, it should send a 0-length content.

File: lib/api/core/entrypoints/embedded/protocols/http.js [hunk 1]
```diff
@@ -262,7 +262,7 @@ class HttpProtocol extends Protocol {
 
     debug('HTTP request response: %a', data);
 
-    if (data.raw) {
+    if (data.raw && data.content) {
```
Comment (alexandrebouthinon) [inline lib/api/core/entrypoints/embedded/protocols/http.js]:
```suggestion
    if (data.raw) {
```

File: lib/api/core/entrypoints/embedded/protocols/http.js [hunk 2]
```diff
@@ -262,7 +262,7 @@ class HttpProtocol extends Protocol {
...
       if (typeof data.content === 'object') {
```
Comment (alexandrebouthinon) [inline lib/api/core/entrypoints/embedded/protocols/http.js]:
```suggestion
      if (data.content && typeof data.content === 'object') {
```

Comment (scottinet) [reply lib/api/core/entrypoints/embedded/protocols/http.js]: not enough: if you do that, you end up stringifying null and returning a string with the word "null" in it

you have to specifically return null if data.content is null, to truly send a 0-length response

File: lib/api/core/entrypoints/embedded/protocols/http.js [hunk 3]
```diff
@@ -263,7 +263,10 @@ class HttpProtocol extends Protocol {
     debug('HTTP request response: %a', data);
 
     if (data.raw) {
-      if (typeof data.content === 'object') {
+      if (!data.content) {
```
Comment (scottinet) [inline lib/api/core/entrypoints/embedded/protocols/http.js]: This test makes Kuzzle return a 0-length response if the content is falsey, such as  the number `0` or the `false` boolean value.

```suggestion
      if (data.content === null || data.content === undefined) {
```

**Comment by ThomasF34** [comment]
Before checking the type of `data.content`, i'm checking if it's null. If it is, we set an header `Content Length` to `0` and set the data to an empty string. Thus, the `_getResponseData` will still return a `Buffer` which is empty if `data.content` is null. 
I also added a unit test to check if the content-length is correctly set to 0

**Pull request merged**